rattan-core/src/cells/per_packet/delay.rs

```diff
@@ -147,13 +146,16 @@ where
         }
 
         // send the packet
-        let (instant, packet) = packet.expect("We cannot exit the previous loop if packet is None");
+        let (instant, mut packet) =
+            packet.expect("We cannot exit the previous loop if packet is None");
         if instant <= Instant::now() {
+            packet.delay_until(instant);
             Some(packet)
         } else {
             // We shouldn't have dequeue the packet, it is not ready to be sent yet.
-            self.packet_queue.renqueue(packet, instant);
-            None
+            // self.packet_queue.renqueue(packet, instant);
+            // None
+            todo!()
```

Comment:
Why you change this to `todo()!`? This seems to be not complete. Or why we can't just return None and reenqueue? Are you assuming this branch will never trigger, otherwise, this will panic at runtime.